### ISSUE TITLE
ALLAUTH_SETTING_GETTER should be a string

### ISSUE THREAD

**Issue by meequz** [pennersr/django-allauth#1854]
I think using `ALLAUTH_SETTING_GETTER` is not properly implemented in `allauth.account.app_settings` and `allauth.socialaccount.app_settings`. Getter should be set as a string, and `import_string` from `django.utils.module_loading` should be used.

How it currently implemented:
```
    def _setting(self, name, dflt):
        from django.conf import settings
        getter = getattr(settings,
                         'ALLAUTH_SETTING_GETTER',
                         lambda name, dflt: getattr(settings, name, dflt))
        return getter(self.prefix + name, dflt)
```

How it should be (just rough example):
```
from django.utils.module_loading import import_string

    def _setting(self, name, dflt):
        from django.conf import settings
        default_getter = lambda name, dflt: getattr(settings, name, dflt)

        custom_getter_string = getattr(settings, 'ALLAUTH_SETTING_GETTER', None)
        if custom_getter_string:
            custom_getter = import_string(custom_getter_string)
        else:
            custom_getter = None

        getter = custom_getter or default_getter
        return getter(self.prefix + name, dflt)
```

Why it should be like that? Because this way I can implement my custom getter in any place of the project. With the current implementation, I need to implement my getter exactly in `settings.py`, which is not very Djangoic. Look at the other settings: almost all of them are strings or numbers. A lot of them are strings for import.

**Comment by pennersr** [comment]
Closed via 31c16afe

**Comment by pennersr** [comment]
https://codeberg.org/allauth/django-allauth/issues/1854
